exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -98,47 +120,70 @@ func newCwLogsExporter(config component.Config, params exp.CreateSettings) (exp.
 }
 
 func (e *exporter) consumeLogs(_ context.Context, ld plog.Logs) error {
-	cwLogsPusher := e.pusher
-	logEvents, _ := logsToCWLogs(e.logger, ld)
+	logEvents, _ := logsToCWLogs(e.logger, ld, e.Config)
 	if len(logEvents) == 0 {
 		return nil
 	}
 
+	var logPushersUsed []cwlogs.Pusher
 	for _, logEvent := range logEvents {
-		logEvent := &cwlogs.Event{
-			InputLogEvent: logEvent,
-			GeneratedTime: time.Now(),
-		}
+		cwLogsPusher := e.getLogPusher(logEvent)
 		e.logger.Debug("Adding log event", zap.Any("event", logEvent))
-		err := cwLogsPusher.AddLogEntry(logEvent)
+		err := cwLogsPusher.AddLogEntry(&logEvent)
 		if err != nil {
 			e.logger.Error("Failed ", zap.Int("num_of_events", len(logEvents)))
 		}
+		logPushersUsed = append(logPushersUsed, cwLogsPusher)
```

Comment:
This will result in adding the same pusher to this list multiple times if it is used multiple times, no?  Should this be a `map[cwlogs.PusherKey]cwlogs.Pusher` to make it a set?

Reply:
Yes, you are right. This will give the pusher multiple times. I will make the changed. It will still work like this because it will just flush an empty pusher for all the duplicates, but is less efficient.